feat: add init container to web deployment for database migrations

chaholl opened the pull request:
## Summary

This PR adds an initialization container to the web deployment to resolve health check timeout issues during startup.

## Problem
The main web container performs database migrations and other initialization tasks on startup, which can take a significant amount of time. This causes health check probes to timeout and the pod to be killed before it can fully start.

## Solution
- Add an init container that runs the full entrypoint script for initialization
- The init container handles database migrations, environment setup, and all initialization logic
- Main container starts only after initialization is complete, allowing it to pass health checks immediately

## Changes
- **Modified:** `charts/langfuse/templates/web/deployment.yaml` - Added init container configuration
- **Added:** `charts/langfuse/tests/init-container_test.yaml` - Comprehensive test coverage

## Key Benefits
- ✅ Prevents health check timeouts during startup
- ✅ Separates initialization concerns from web serving
- ✅ Runs full entrypoint script with proper error handling
- ✅ Preserves existing `extraInitContainers` functionality
- ✅ Fully tested with no regressions

## Testing
- All existing tests pass (50/50)
- New comprehensive test coverage for init container functionality
- Helm linting passes without issues

Closes #272
<!-- ELLIPSIS_HIDDEN -->

----

> [!IMPORTANT]
> Adds an init container to `deployment.yaml` for database migrations, preventing health check timeouts, with tests in `init-container_test.yaml`.
> 
>   - **Behavior**:
>     - Adds init container `langfuse-web-init` to `deployment.yaml` for database migrations and initialization tasks.
>     - Main container starts post-initialization, preventing health check timeouts.
>   - **Configuration**:
>     - Init container runs full entrypoint script and exits with `echo "Init completed successfully"`.
>     - Uses same image as main container and includes environment variables.
>     - Preserves `extraInitContainers` functionality.
>   - **Testing**:
>     - Adds `init-container_test.yaml` for testing init container configuration and functionality.
>     - Tests ensure init container exists, uses correct image, and handles custom init containers.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=langfuse%2Flangfuse-k8s&utm_source=github&utm_medium=referral)<sup> for ee806f078b0b7a065b6dd92452731dc0bd469da6. You can [customize](https://app.ellipsis.dev/langfuse/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

<!-- ELLIPSIS_HIDDEN -->

chaholl committed:
feat: add init container to web deployment for database migrations

- Add init container to handle database migrations and initialization
- Prevents health check timeouts by separating initialization from web serving
- Init container runs the full entrypoint script then exits cleanly
- Preserves existing extraInitContainers functionality
- Add comprehensive test coverage for init container functionality

Resolves startup issues where health checks were failing due to
time-consuming database migrations running in the main container.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/langfuse/langfuse-k8s?pullRequest=274) <br/>All committers have signed the CLA.

chaholl committed:
fix: use args instead of command for init container

- Changed from 'command' to 'args' to allow entrypoint script to run
- This ensures the full initialization logic executes properly
- The entrypoint script handles environment setup, DB migrations, etc.
- Container exits cleanly after running 'echo Init completed successfully'

chaholl committed:
fix: increase liveness probe initial delay to 60s

- Update default initialDelaySeconds from 20s to 60s in both values.yaml and deployment template
- Gives main container more time to start after init container completes
- Prevents premature health check failures during startup

Steffen911 commented:
@chaholl We wondered whether chart hooks are a more applicable function here to schedule a single upgrade job before updating the containers. The init container may still suffer from race conditions as each pod attempts to start simultaneously, i.e. it would only address parts of the problem.
What do you think about those approaches?

An additional step as part of this PR could be too actively disable migrations in the main container, but that's probably not necessary as they'll complete quickly.

chaholl committed:
docs: update chart documentation for liveness probe changes

- Auto-generated documentation using helm-docs
- Reflects updated initialDelaySeconds from 20s to 60s

Steffen911 commented:
> For sure, chart hooks would be much nicer. But we're still running entrypoint.sh on main container start. We'd need to 'fix' that to really gain the benefit of running an upgrade job vs multiple init containers.

If we were to set `LANGFUSE_AUTO_CLICKHOUSE_MIGRATION_DISABLED=true` and `LANGFUSE_AUTO_POSTGRES_MIGRATION_DISABLED=true` the migrations should be skipped within the entrypoint. In that case, the central init job should work well.

chaholl commented:
let me test that with the original 20s value and we'll see how it goes

Steffen911 commented:
@chaholl Thank you so much! Let me know how it goes.
If this doesn't have a large benefit, I'll be happy to test and merge this PR early next week as this is a solid improvement already.

chaholl commented:
Had a look into this. Init containers are the only option without causing a world of pain! 

The problem with chart hooks is that they'll run either before we install the rest of the chart, or afterwards. Before is too soon because the databases aren't deployed, after is too late because the web has already been deployed. It would be easy enough if there were no subcharts, we'd just set the hook-weights, but with subcharts, it depends on what's exposed as values and whether they use hooks internally.

If we stick with the init containers, [the locking mechanism in prima should prevent trouble](https://www.prisma.io/docs/orm/prisma-migrate/workflows/development-and-production#advisory-locking)

gileri commented on charts/langfuse/values.yaml:
```diff
@@ -262,7 +262,7 @@ langfuse:
       path: "/api/public/health"
       # -- Initial delay seconds for livenessProbe.
-      initialDelaySeconds: 20
+      initialDelaySeconds: 60
```
That is my experience with this chart too.
